From 885d52ae38adb2cd721ab6501f5cdc929d0bbcfe Mon Sep 17 00:00:00 2001
From: Michael Teyfel <mteyfel@de.adit-jv.com>
Date: Thu, 21 Dec 2017 16:10:13 +0100
Subject: [PATCH 3/3] ivi-controller: load id-agent module

Signed-off-by: Michael Teyfel <mteyfel@de.adit-jv.com>
---
 weston-ivi-shell/src/ivi-controller.c | 36 +++++++++++++++++++++++++++++++++++
 1 file changed, 36 insertions(+)

diff --git a/weston-ivi-shell/src/ivi-controller.c b/weston-ivi-shell/src/ivi-controller.c
index 8faec2a..fd6c7d8 100644
--- a/weston-ivi-shell/src/ivi-controller.c
+++ b/weston-ivi-shell/src/ivi-controller.c
@@ -2170,6 +2170,38 @@ launch_client_process(void *data)
     free(shell->ivi_client_name);
 }
 
+static int load_id_agent_module(struct ivishell *shell)
+{
+    struct weston_config *config = wet_get_config(shell->compositor);
+    struct weston_config_section *section;
+    char *id_agent_module = NULL;
+
+    int (*id_agent_module_init)(struct weston_compositor *compositor,
+            const struct ivi_layout_interface *interface);
+
+    section = weston_config_get_section(config, "ivi-shell", NULL, NULL);
+
+    if (weston_config_section_get_string(section, "ivi-id-agent-module",
+                                         &id_agent_module, NULL) < 0) {
+        /* input events are handled by weston's default grabs */
+        weston_log("ivi-controller: No ivi-id-agent-module set\n");
+        return 0;
+    }
+
+    id_agent_module_init = wet_load_module_entrypoint(id_agent_module, "id_agent_module_init");
+    if (!id_agent_module_init)
+        return -1;
+
+    if (id_agent_module_init(shell->compositor, shell->interface) != 0) {
+        weston_log("ivi-controller: Initialization of id-agent module failed\n");
+        return -1;
+    }
+
+    free(id_agent_module);
+
+    return 0;
+}
+
 WL_EXPORT int
 wet_module_init(struct weston_compositor *compositor,
 		       int *argc, char *argv[])
@@ -2226,5 +2258,9 @@ wet_module_init(struct weston_compositor *compositor,
         wl_event_loop_add_idle(loop, launch_client_process, shell);
     }
 
+    if (load_id_agent_module(shell) < 0) {
+        weston_log("ivi-controller: id-agent module not loaded\n");
+    }
+
     return 0;
 }
-- 
2.7.4

