From 6a6ee63a3a0b1bd0756face89bea2d8e2dfe4366 Mon Sep 17 00:00:00 2001
From: Oleksandr Tyshchenko <oleksandr_tyshchenko@epam.com>
Date: Thu, 15 Mar 2018 12:41:30 +0200
Subject: [PATCH 1/2] Enable everything needed source command to work

Signed-off-by: Oleksandr Tyshchenko <oleksandr_tyshchenko@epam.com>
---
 include/configs/rcar-gen3-common.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/configs/rcar-gen3-common.h b/include/configs/rcar-gen3-common.h
index 24ce2de..0bef2e7 100644
--- a/include/configs/rcar-gen3-common.h
+++ b/include/configs/rcar-gen3-common.h
@@ -125,4 +125,8 @@
 #define CONFIG_SYS_MALLOC_LEN		(1 * 1024 * 1024)
 #define CONFIG_SYS_BOOTMAPSZ		(8 * 1024 * 1024)
 
+#define CONFIG_CMD_SOURCE
+#define CONFIG_CMD_ECHO
+#define CONFIG_IMAGE_FORMAT_LEGACY /* enable also legacy image format */
+
 #endif	/* __RCAR_GEN3_COMMON_H */
-- 
2.7.4

