From a4da9161a95d68aa934b7085d8e55c9f68c7c98c Mon Sep 17 00:00:00 2001
From: Michael Teyfel <mteyfel@de.adit-jv.com>
Date: Mon, 6 Nov 2017 12:55:07 +0100
Subject: [PATCH 08/15] ivi-layout: use libweston-desktop api for views

Signed-off-by: Michael Teyfel <mteyfel@de.adit-jv.com>
---
 ivi-shell/ivi-layout.c | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/ivi-shell/ivi-layout.c b/ivi-shell/ivi-layout.c
index 4247041..d852d62 100644
--- a/ivi-shell/ivi-layout.c
+++ b/ivi-shell/ivi-layout.c
@@ -153,7 +153,10 @@ ivi_view_destroy(struct ivi_layout_view *ivi_view)
 	wl_list_remove(&ivi_view->pending_link);
 	wl_list_remove(&ivi_view->order_link);
 
-	weston_view_destroy(ivi_view->view);
+	if (weston_surface_is_desktop_surface(ivi_view->ivisurf->surface))
+		weston_desktop_surface_unlink_view(ivi_view->view);
+	else
+		weston_view_destroy(ivi_view->view);
 
 	free(ivi_view);
 }
@@ -170,7 +173,13 @@ ivi_view_create(struct ivi_layout_layer *ivilayer,
 		return NULL;
 	}
 
-	ivi_view->view = weston_view_create(ivisurf->surface);
+	if (weston_surface_is_desktop_surface(ivisurf->surface)) {
+		ivi_view->view = weston_desktop_surface_create_view(
+				ivisurf->weston_desktop_surface);
+	} else {
+		ivi_view->view = weston_view_create(ivisurf->surface);
+	}
+
 	if (ivi_view->view == NULL) {
 		weston_log("fails to allocate memory\n");
 		free(ivi_view);
-- 
2.7.4

